src/metamath.c

```diff
@@ -742,14 +743,12 @@ int main(int argc, char *argv[]) {
 /* argc is the number of arguments; argv points to an array containing them */
 
 # ifdef RUN_REGRESSION_TEST
+// activated if at least one unit requests regression tests.
+// a failing test MUST terminate the program with exit(EXIT_FAILURE).
```

Comment:
If a test terminates the program, how will the rest of the tests run? I think it should instead either throw an exception (not an option in C AFAIK) or return failure so that the test harness can aggregate information on all successes and failures.

Reply:
If your modifications, say, broke a primitive routine, then all tests based on this primitive are likely to fail as well.  It is pointless to watch such a cascading effect in total.  Compilers follow a similar strategy.  A missing brace may entail hundreds of grammar errors in the following code lines, but usually only the first 10 or so are shown, before the process stops.

It is a matter of taste, or, if you like, minor improvement, to not stop at the first error, but continue up to the n-th one.  I don't mind if you, or someone else, modifies the process in that direction.  But please, do not expect me to do this optimization.

See, after about 3 weeks this PR contains the allocation and initialization of a buffer.  That's it.  About 6 lines of my own code wait to pass the review, because I am quite some time bothered with creating an infrastructure, that wasn't in place before, and that, strictly speaking, is not necessary for my primary goal.  We need to make some progress on the core ideas as well.

Reply:
You previously had an implementation that would do exactly this (with the `ok` variable that was being assigned in the harness), I'm not sure why you removed it.

It's true that tests can often fail in batches. But not all bugs are like that, and if the test suite has good coverage then hopefully they will be more orthogonal than that. I think the simpler thing is just to let them all run and let the logs have the information necessary.

You seem to be frustrated that it is taking a while to add this test suite, but I believe the idea of a regression test suite was your idea. If you are going to do a test suite, please take the time to make it good, or don't bother in the first place and use the test suite we already have.

Reply:
I added my personal hand-written ad-hoc code in #86 to ease maintenance of this piece of software.  For one I wanted to make sure I got everything right.  And then I wanted to be prepared if people request changes.  The test code was actually used in both cases, and proved useful.  It was hidden behind a ifdef .. endif clause and was so never part of a regular build.  It could simply have been ignored.

My expectations was that after some initial discussions modifications peter out and the software won't be touched for years again.  In this phase the test code could have been safely dropped.

In hindsight I should have kept the test code on my computer and simply included it in my private build each time someone plays around with the code.  And complain if the tests fail.

I am not opposed to provide people with a regression test infrastructure.  I think it is a MUST for mathematical software like this, because it becomes part of the reasoning and replaces peer review.  So confidence is an issue.  And my review on pending PR from others show me how important this is to asses contributions of others.

But I am opposed to work on a test framework beyond the scope needed for this PR.  By the way, I think another infrastructure layer supporting debugging in a systematic way is necessary, too (Won't open this can of worms, though).

Reply:
> I'm not sure why you removed it

Implementations of outside test frameworks suggested this, and I followed the pattern.  One obvious result is the code lines reduce, because all is done in a macro call.